addons/web/static/lib/hoot-dom/helpers/events.js

```diff
@@ -0,0 +1,1811 @@
+        }
+    }
+
+    return logEvents("leave");
+}
+
+/**
+ * Shorthand helper to attach an event listener to the given {@link Target}, and
+ * returning a function to remove the listener.
+ *
+ * @template {EventType} T
+ * @param {EventTarget | Target} target
+ * @param {T} type
+ * @param {(event: GlobalEventHandlersEventMap[T]) => any} listener
+ * @param {boolean | AddEventListenerOptions} [options]
+ * @returns {() => void}
+ * @example
+ *  const off = on("button", "click", onClick);
+ *  after(off);
+ */
+export function on(target, type, listener, options) {
+    const targets = isEventTarget(target) ? [target] : queryAll(target);
+    if (!targets.length) {
+        throw new HootDomError(`expected at least 1 event target, got none`);
+    }
+    for (const eventTarget of targets) {
+        eventTarget.addEventListener(type, listener, options);
+    }
+
+    return function off() {
+        for (const eventTarget of targets) {
+            eventTarget.removeEventListener(type, listener, options);
+        }
+    };
+}
+
+/**
+ * Performs a pointer down on the given {@link Target}.
+ *
+ * The event sequence is as follow:
+ *  - `pointerdown`
+ *  - [desktop] `mousedown`
+ *  - [touch] `touchstart`
+ *  - [target is not active element] `blur`
+ *  - [target is focusable] `focus`
+ *
+ * @param {Target} target
+ * @param {QueryOptions} [options]
+ * @returns {Event[]}
+ * @example
+ *  pointerDown("button"); // Focuses to the first <button> element
+ */
+export function pointerDown(target, options) {
+    const element = getFirstTarget(target, options);
+    const pointerInit = {
+        ...getPosition(element, options),
+        button: options?.button || 0,
+    };
+    _pointerDown(element, pointerInit);
+
+    return logEvents("pointerDown");
+}
+
+/**
+ * Performs a pointer up on the given {@link Target}.
+ *
+ * The event sequence is as follow:
+ * - `pointerup`
+ * - [desktop] `mouseup`
+ * - [touch] `touchend`
+ *
+ * @param {Target} target
+ * @param {QueryOptions} [options]
+ * @returns {Event[]}
+ * @example
+ *  pointerUp("body"); // Triggers a pointer up on the <body> element
+ */
+export function pointerUp(target, options) {
+    const element = getFirstTarget(target, options);
+    const pointerInit = {
+        ...getPosition(element, options),
+        button: options?.button || 0,
+    };
+    _pointerUp(element, pointerInit);
+
+    return logEvents("pointerUp");
+}
+
+/**
+ * Performs a keyboard event sequence on the given {@link Target}.
+ *
+ * The event sequence is as follow:
+ *  - `keydown`
+ *  - `keypress`
+ *  - `keyup`
+ *
+ * @param {KeyStrokes} keyStrokes
+ * @returns {Event[]}
+ * @example
+ *  pointerDown("button[type=submit]"); // Moves focus to <button>
+ *  keyDown("Enter"); // Submits the form
+ * @example
+ *  keyDown("Shift+Tab"); // Focuses previous focusable element
+ * @example
+ *  keyDown(["Ctrl", "v"]); // Pastes current clipboard content
+ */
+export function press(keyStrokes) {
```

Comment:
press("Control+[somekey]") doesn't work as expected. Since the _press fonction does a keyDown and a keyUp right after ,the modifier ctrKey is never set on the press of [someKey]